src/game/SpellEffects.cpp

```diff
@@ -3638,23 +3638,41 @@ void Spell::DoSummon(SpellEffectIndex eff_idx)
     if (!pet_entry)
         return;
 
+    Pet* spawnCreature = new Pet();
+
+    // set timer for unsummon
+    if (m_duration > 0)
+        spawnCreature->SetDuration(m_duration);
+
+    if (m_caster->GetTypeId() == TYPEID_PLAYER)
+    {
+        // Load pet from db; if any to load
+        if (m_caster->getClass() == CLASS_MAGE && spawnCreature->LoadPetFromDB((Player*)m_caster, pet_entry)) 
```

Comment:
I have no idea honestly - At least I'm pretty sure the druid treants are NOT controllable, but I am not even 100% sure on that one, I switched from druid to rogue from retail vanilla to tbc..